ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                    MM_strnzcpy(
+                        phMiraMonLayer->pLayerDB->pFields[iField].pszFieldName,
+                        pszString, MM_MAX_LON_FIELD_NAME_DBF);
+                    CPLFree(pszString);
+                }
+                else
+                {
+                    MM_strnzcpy(
+                        phMiraMonLayer->pLayerDB->pFields[iField].pszFieldName,
+                        poFeatureDefn->GetFieldDefn(iField)->GetNameRef(),
+                        MM_MAX_LON_FIELD_NAME_DBF);
+                }
+            }
+
+            if (poFeatureDefn->GetFieldDefn(iField)->GetAlternativeNameRef())
+            {
+                if (phMiraMonLayer->nCharSet != MM_JOC_CARAC_UTF8_DBF)
+                {
+                    char *pszString =
+                        CPLRecode(poFeatureDefn->GetFieldDefn(iField)
+                                      ->GetAlternativeNameRef(),
+                                  CPL_ENC_UTF8, CPL_ENC_ISO8859_1);
+                    MM_strnzcpy(phMiraMonLayer->pLayerDB->pFields[iField]
+                                    .pszFieldDescription,
+                                pszString, MM_MAX_BYTES_FIELD_DESC);
+                    CPLFree(pszString);
+                }
+                else
+                {
+                    MM_strnzcpy(phMiraMonLayer->pLayerDB->pFields[iField]
+                                    .pszFieldDescription,
+                                poFeatureDefn->GetFieldDefn(iField)
+                                    ->GetAlternativeNameRef(),
+                                MM_MAX_BYTES_FIELD_DESC);
+                }
+            }
+            phMiraMonLayer->pLayerDB->nNFields++;
+        }
+    }
+
+    CPLDebug("MiraMon", "Fields to MiraMon translated.");
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                       TranslateFieldsValuesToMM()                        */
+/*                                                                          */
+/*      Translase ogr Fields to a structure that MiraMon can understand     */
+/****************************************************************************/
+
+OGRErr OGRMiraMonLayer::TranslateFieldsValuesToMM(OGRFeature *poFeature)
+
+{
+    if (poFeatureDefn->GetFieldCount() == 0)
+    {
+        // MiraMon have private DataBase records
+        hMMFeature.nNumMRecords = 1;
+        return OGRERR_NONE;
+    }
+
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nIRecord;
+    int nNumFields = poFeatureDefn->GetFieldCount();
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nNumRecords;
+    hMMFeature.nNumMRecords = 0;
+
+    for (int iField = 0; iField < nNumFields; iField++)
+    {
+        OGRFieldType eFType = poFeatureDefn->GetFieldDefn(iField)->GetType();
+        const char *pszRawValue = poFeature->GetFieldAsString(iField);
+
+        if (eFType == OFTStringList)
+        {
+            char **panValues = poFeature->GetFieldAsStringList(iField);
```

Comment:
This should be papszValues here  (panValues stands for (p)ointer to an (a)rray of i(n)teger values)

Reply:
Done